modules/host/network.go

```diff
@@ -174,6 +161,20 @@ func (h *Host) threadedListen() {
 	}
 	defer h.tg.Done()
 
+	// Automatically close the listener when h.tg.Stop() is called.
+	err = h.tg.Add()
+	if err != nil {
+		return
+	}
+	go func() {
+		defer h.tg.Done()
+		<-h.tg.StopChan()
+		err = h.listener.Close()
+		if err != nil {
+			h.log.Println("Closing the listener failed:", err)
+		}
+	}()
+
```

Comment:
This does not re-introduce the bug as far as I can tell. Can you explain your reasoning more?

This is in contrast to the paradigm that every parent should be responsible for killing it's children. This paradigm works well when the children never outlive the parent, but that is not true here. When the children outlive the parent I think a better paradigm is for the thread to be responsible for killing itself.